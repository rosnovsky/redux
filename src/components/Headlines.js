import React from 'react';
import '../styles/App.css';

function Headlines() {
    // as soon as we get news articles in redux store, we will dispatch an action SELECT_HEADLINES, which will return an array of selected headlines (say, with word Linux in them; this is completely unnecessary and normally we'd do this on the API level, but hey, it's a workout, so let's fo more then we have to.

    // Whenever a user clicks on a headline, we will dispatch an action ACTIVATE_ARTICLE, and this action will put current article and all its metadata into the state object for other components to consume.

    return (
        <div className="App">
            <h2>Headlines</h2>
        </div>
    );
}

export default Headlines;
