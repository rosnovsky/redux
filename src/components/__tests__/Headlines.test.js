import React from 'react';
import { create } from 'react-test-renderer';
import Headlines from '../Headlines';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { mount } from 'enzyme';

Enzyme.configure({ adapter: new Adapter() });

describe('Headlines', () => {
    const articles = [
        { title: 'First Article', id: '1' },
        { title: 'Second Article', id: '2' },
        { title: 'Third Article', id: '3' },
    ];
    const initialState = { articles, activeArticle: null };

    it('Headlines component renders without crashing', () => {
        const app = create(mount(<Headlines articles={articles} />));
        let tree = app.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Headlines IDs work correctly', () => {
        const app = create(mount(<Headlines articles={articles} />));
        expect(app.toJSON().children[2].props.id).toBe('2');
    });

    it('Headlines render correctly', () => {
        const app = mount(<Headlines articles={articles} />);
        expect(app.find('h3')).toHaveLength(3);
    });
});
