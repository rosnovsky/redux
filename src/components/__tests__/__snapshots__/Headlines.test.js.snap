// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headlines Headlines component renders without crashing 1`] = `
<div
  className="App"
>
  <h2>
    Headlines
  </h2>
  <h3
    id="1"
    onClick={[Function]}
    style={
      Object {
        "color": "black",
      }
    }
  >
    First Article
  </h3>
  <h3
    id="2"
    onClick={[Function]}
    style={
      Object {
        "color": "black",
      }
    }
  >
    Second Article
  </h3>
  <h3
    id="3"
    onClick={[Function]}
    style={
      Object {
        "color": "black",
      }
    }
  >
    Third Article
  </h3>
</div>
`;
